kdf/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+#![doc = include_str!("../README.md")]
+#![no_std]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/RustCrypto/media/8f1a9894/logo.svg",
+    html_favicon_url = "https://raw.githubusercontent.com/RustCrypto/media/8f1a9894/logo.svg"
+)]
+#![forbid(unsafe_code)]
+#![warn(missing_docs, unused_qualifications, missing_debug_implementations)]
+
+/// Use the KDF to derive the given amount of output.
+pub trait Derive {
```

Comment:
Perhaps this should be `Extract`?
```suggestion
pub trait Extract {
```

Reply:
I guess one question is: does the "extract" name make sense for KDFs outside an extract-and-expand context?

Reply:
Maybe it's better to call it `KeyDeriveFn` and name the method `derive_key`? Though the `Fn` part is a bit problematic since it may be confused with the `Fn` closure traits.